test_runner/regress/test_tenant_size.py

```diff
@@ -532,7 +532,24 @@ def get_current_consistent_size(
     assert size_after == prev, "size after restarting pageserver should not have changed"
 
 
-@xfail_on_postgres(PgVersion.V15, reason="Test significantly more flaky on Postgres 15")
+def assert_approx_equal(size_a, size_b):
+    """
+    Tests that evaluate sizes are checking the pageserver space consumption
+    that sits many layers below the user input.  The exact space needed
+    varies slightly depending on postgres behavior.
+
+    Rather than expecting postgres to be determinstic and occasionally
+    failing the test, we permit sizes for the same data to vary by a few pages.
+    """
+
+    # Determined empirically from examples of equality failures: they differ
+    # by page multiples of 8272, and usually by 1-3 pages.
+    threshold = 4 * 8272
+
+    if abs(size_a - size_b) > threshold:
+        raise AssertionError(f"Failed {size_a} != {size_b} (exceeded {threshold})")
```

Comment:
There's [`pytest.approx`](https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest-approx) method, we can internally use (or inline it and move the threshold to a constant, up to you)

Reply:
I've tweaked this to use pytest.approx -- kind of suprised they thought that was worth a helper, but I guess it gives more consistently formatted assertion messages.